controllers/schedule_it_test.go

```diff
@@ -0,0 +1,70 @@
+// +build integration
+
+package controllers_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	k8upv1alpha1 "github.com/vshn/k8up/api/v1alpha1"
+	"github.com/vshn/k8up/controllers"
+)
+
+type (
+	ScheduleControllerTestSuite struct {
+		EnvTestSuite
+		reconciler    *controllers.ScheduleReconciler
+		givenSchedule *k8upv1alpha1.Schedule
+	}
+)
+
+func Test_Schedule(t *testing.T) {
+	suite.Run(t, new(ScheduleControllerTestSuite))
+}
+
+func (ts *ScheduleControllerTestSuite) BeforeTest(suiteName, testName string) {
+	ts.reconciler = &controllers.ScheduleReconciler{
+		Client: ts.Client,
+		Log:    ts.Logger.WithName(suiteName),
+		Scheme: ts.Scheme,
+	}
+}
+
+func (ts *ScheduleControllerTestSuite) Test_GivenScheduleWithRandomSchedules_WhenChangingToStandardSchedule_ThenCleanupEffectiveSchedule() {
+	ts.givenScheduleResource()
+
+	ts.whenReconciling(ts.givenSchedule)
+
+	ts.thenAssertEffectiveScheduleExists(ts.givenSchedule.Name)
+
+	resultSchedule := &k8upv1alpha1.Schedule{}
+	ts.FetchResource(k8upv1alpha1.GetNamespacedName(ts.givenSchedule), resultSchedule)
+	ts.thenAssertCondition(resultSchedule, k8upv1alpha1.ConditionReady, k8upv1alpha1.ReasonReady, "resource is ready")
+
+	// Change schedule spec
+	resultSchedule.Spec.Backup.Schedule = "* * * * *"
+	ts.UpdateResources(resultSchedule)
+
+	ts.whenReconciling(resultSchedule)
+
+	resultSchedule = ts.newScheduleSpec(ts.givenSchedule.Name)
+	ts.FetchResource(k8upv1alpha1.GetNamespacedName(resultSchedule), resultSchedule)
+	ts.thenAssertCondition(resultSchedule, k8upv1alpha1.ConditionReady, k8upv1alpha1.ReasonReady, "resource is ready")
+
+	newEffectiveSchedules := ts.whenListEffectiveSchedules()
```

Comment:
Perhaps `actualEffectiveSchedules` could be a better term, as it _actual_ is a term often used to compare the actual state with the expected state.